test/plugins/Collateral.test.ts

```diff
@@ -622,29 +625,153 @@ describe('Collateral contracts', () => {
+      await expectUnpriced(cTokenCollateral.address)
 
       // When refreshed, sets status to Unpriced
       await tokenCollateral.refresh()
+      await aTokenCollateral.refresh()
+      await cTokenCollateral.refresh()
+
       expect(await tokenCollateral.status()).to.equal(CollateralStatus.IFFY)
+      expect(await aTokenCollateral.status()).to.equal(CollateralStatus.IFFY)
+      expect(await cTokenCollateral.status()).to.equal(CollateralStatus.IFFY)
+    })
+
+    it('Should be able to refresh saved prices', async () => {
+      // Check one Fiat Collateral and one Appreciating Collateral
+      const collsToProcess = [tokenCollateral, aTokenCollateral]
+      const initialBlockTimestamp: number = await getLatestBlockTimestamp()
+
+      for (const coll of collsToProcess) {
+        // Check initial prices
+        await expectPrice(coll.address, fp('1'), ORACLE_ERROR, true)
+        let [lowPrice, highPrice] = await coll.price()
+        expect(await coll.savedLowPrice()).to.equal(lowPrice)
+        expect(await coll.savedHighPrice()).to.equal(highPrice)
+        expect(await coll.lastSave()).to.equal(initialBlockTimestamp)
+
+        // Refresh saved prices
+        await coll.refresh()
+
+        // Check values remain but timestamp was updated
+        await expectPrice(coll.address, fp('1'), ORACLE_ERROR, true)
+        ;[lowPrice, highPrice] = await coll.price()
+        expect(await coll.savedLowPrice()).to.equal(lowPrice)
+        expect(await coll.savedHighPrice()).to.equal(highPrice)
+        let currBlockTimestamp = await getLatestBlockTimestamp()
+        expect(await coll.lastSave()).to.equal(currBlockTimestamp)
+
+        // Update values in Oracles increase by 20%
+        await setOraclePrice(coll.address, bn('1.2e8')) // 20%
+
+        // Before calling refresh we still have the old values
+        await expectPrice(coll.address, fp('1.2'), ORACLE_ERROR, true)
+        ;[lowPrice, highPrice] = await coll.price()
+        expect(await coll.savedLowPrice()).to.be.lt(lowPrice)
+        expect(await coll.savedHighPrice()).to.be.lt(highPrice)
+
+        // Refresh prices - Should save new values
+        await coll.refresh()
+
+        // Check new prices were stored
+        await expectPrice(coll.address, fp('1.2'), ORACLE_ERROR, true)
+        ;[lowPrice, highPrice] = await coll.price()
+        expect(await coll.savedLowPrice()).to.equal(lowPrice)
+        expect(await coll.savedHighPrice()).to.equal(highPrice)
+        currBlockTimestamp = await getLatestBlockTimestamp()
+        expect(await coll.lastSave()).to.equal(currBlockTimestamp)
+
+        // Restore value in Oracle for next collateral
+        await setOraclePrice(coll.address, bn('1e8'))
+      }
+    })
+
+    it('Should not save prices if try/price returns unpriced - Appreciating Collateral', async () => {
+      const UnpricedAppreciatingFactory = await ethers.getContractFactory(
+        'UnpricedAppreciatingFiatCollateralMock'
```

Comment:
I think we could delete `UnpricedAppreciatingFiatCollateralMock` contract and just set oracle price to something out of range.

Reply:
Unfortunately we do need it because it is testing this line: https://github.com/reserve-protocol/protocol/blob/66b30dd20715838acc7db29a8d5df5212cdc2143/contracts/plugins/assets/AppreciatingFiatCollateral.sol#L117

There is a difference between having `try.price` fail and having a Collateral that returns [0,FIX_MAX] but returns the price without reverting. This is the case im covering which was only covered for `FiatCollateral`.  

Unless we want to remove that condition in the Plugin or leave it uncovered.

Reply:
Fun fact: The one that covers it for `FiatCollateral` is the `EUR Collateral` which returns [0, FIX_MAX] if the 2nd oracle returns 0.

Reply:
Gotcha, that is different. I see.